code/modules/surgery/organs/organ.dm

```diff
@@ -296,14 +287,15 @@ var/list/organ_cache = list()
 	if(!istype(owner))
 		return
 
-	var/obj/item/organ/internal/O = owner.get_int_organ_tag(organ_tag)
-	world << "[O]"
-	if(O)
-		owner.internal_organs -= O
-		//O.forceMove(src)
+	//if(is_primary_organ())//Toddo from fethas:Do i need to move ths?
+	//	owner.internal_organs_by_name[organ_tag] = null
```

Comment:
OOops I think this might accidentally revert some of your code - watch out for this